sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/TimeSeriesLiveTests.cs

```diff
@@ -138,10 +138,7 @@ await foreach (MetricSeriesData seriesData in client.GetMetricSeriesDataAsync(Me
                 returnedKeys.Add(seriesData.Definition.SeriesKey);
             }
 
-            IEnumerable<List<KeyValuePair<string, string>>> expectedKvps = seriesKeys.Select(key => key.AsDictionary().ToList());
-            IEnumerable<List<KeyValuePair<string, string>>> returnedKvps = returnedKeys.Select(key => key.AsDictionary().ToList());
-
-            Assert.That(returnedKvps, Is.EquivalentTo(expectedKvps));
+            Assert.That(seriesKeys, Is.EquivalentTo(returnedKeys));
```

Comment:
In general, it's not expected to work when comparing reference types (except if you really expect the collections to have the same references). We can use it with `DimensionKey` because it has a custom `Equals` method.